### PR TITLE
nrf_security: PAKE support

### PR DESCRIPTION
**Pull request by vili-nordic**

### CODE REVIEW

File: nrf_security/configs/psa_crypto_config.h.template
Comment (joerchan) [inline nrf_security/configs/psa_crypto_config.h.template:283]: PSA_CRYPTO_DRIVER_ALG_ for all of these.

File: nrf_security/include/psa/crypto_driver_contexts_kdf.h
Comment (joerchan) [inline nrf_security/include/psa/crypto_driver_contexts_kdf.h:45]: nit: `#if defined(` for consistency?

File: nrf_security/src/drivers/nrf_oberon/Kconfig
Comment (joerchan) [inline nrf_security/src/drivers/nrf_oberon/Kconfig:278]: Add `depends on PSA_CORE_OBERON` for these three.
I don't think this would compile if you enabled `PSA_CORE_BUILTIN`.

File: nrf_security/src/drivers/nrf_oberon/CMakeLists.txt
Comment (joerchan) [inline nrf_security/src/drivers/nrf_oberon/CMakeLists.txt:40]: These three needs to excluded when not enabled. Since they probably won't compile when PSA_CORE_BUILTIN is enabled. The same issue already exists for oberon_ctr/hmac_drbg.

The macro append_with_prefix_ifdef is broken, but after this PR is merged you can use that one:
https://github.com/nrfconnect/sdk-nrfxlib/pull/971/files

File: nrf_security/src/drivers/nrf_oberon/Kconfig
Comment (joerchan) [inline nrf_security/src/drivers/nrf_oberon/Kconfig:271]: `depends on PSA_CORE_OBERON` also

File: nrf_security/src/psa_crypto_driver_wrappers.c
Comment (joerchan) [inline nrf_security/src/psa_crypto_driver_wrappers.c:2909]: Please ifdef out all the new stuff with `#if defined(CONFIG_PSA_OBERON_CORE)`

**Review by joerchan — CHANGES_REQUESTED** [review]
Looks good, just need some changes for PSA_CORE_BUILTIN to not generate build warnings or errors.

File: nrf_security/src/drivers/nrf_oberon/Kconfig
Comment (Vge0rge) [inline nrf_security/src/drivers/nrf_oberon/Kconfig:337]: Is the SRP the only experimental? Isn't it the same for jpake  and spake etc?

File: nrf_security/src/psa_crypto_driver_wrappers.c
Comment (Vge0rge) [inline nrf_security/src/psa_crypto_driver_wrappers.c:2918]: Nit: Since the pake  function are not standardized we are writing the functions in a different way here and we ifdef algorithms and not drivers. Maybe just write a small comment saying this.

File: nrf_security/src/drivers/nrf_oberon/Kconfig
Comment (vili-nordic) [reply nrf_security/src/drivers/nrf_oberon/Kconfig:337]: I'll add it for the others

File: nrf_security/src/drivers/nrf_oberon/Kconfig
Comment (Vge0rge) [inline nrf_security/src/drivers/nrf_oberon/Kconfig:271]: Experimental here as well probably?

**Review by joerchan — APPROVED** [review]

File: nrf_security/src/psa_crypto_driver_wrappers.c
Comment (frkv) [inline nrf_security/src/psa_crypto_driver_wrappers.c:148]: Possibly NIT: This driver-id is also used for Builtin Mbed TLS

Comment (frkv) [reply nrf_security/src/psa_crypto_driver_wrappers.c:148]: Discussed offline, changing this is too much work for this PR

**Review by frkv — APPROVED** [review]
LGTM

**Review by Vge0rge — APPROVED** [review]

**Review by SebastianBoe — CHANGES_REQUESTED** [review]
We have decided to block this on the move to nrf

**Comment by vili-nordic** [comment]
nrf_security has been moved to nrf

**Pull request closed without merging**
